src/services/transactionService.ts

```diff
@@ -105,6 +118,57 @@ export const validateTransactionParams = (
   return null;
 };
 
+/**
+ * Validates swap transaction parameters
+ * Returns an error message if any validation fails
```

Comment:
Will those error messages be displayed to the user? If so, I think we should translate them